File: src/main/java/com/leijendary/spring/boot/template/core/security/Encryption.java
package com.leijendary.spring.boot.template.core.security;

import lombok.RequiredArgsConstructor;
import org.springframework.cloud.bootstrap.encrypt.KeyProperties;
import org.springframework.security.crypto.encrypt.TextEncryptor;
import org.springframework.stereotype.Component;

import static org.springframework.security.crypto.encrypt.Encryptors.delux;

@Component
@RequiredArgsConstructor
public class Encryption {

    private final KeyProperties keyProperties;

    public String encrypt(final String raw) {
        return encryptor().encrypt(raw);
    }

    public String decrypt(final String encrypted) {
        return encryptor().decrypt(encrypted);
    }

    private TextEncryptor encryptor() {
        final var key = keyProperties.getKey();
        final var salt = keyProperties.getSalt();

        return delux(key, salt);
    }
}

File: src/main/kotlin/com/leijendary/spring/core/security/Encryption.kt
package com.leijendary.spring.core.security

import org.springframework.cloud.bootstrap.encrypt.KeyProperties
import org.springframework.security.crypto.encrypt.Encryptors.delux
import org.springframework.security.crypto.encrypt.TextEncryptor
import org.springframework.stereotype.Component

@Component
class Encryption(private val keyProperties: KeyProperties) {
    fun encrypt(raw: String?): String {
        return encryptor().encrypt(raw)
    }

    fun decrypt(encrypted: String?): String {
        return encryptor().decrypt(encrypted)
    }

    private fun encryptor(): TextEncryptor {
        return delux(keyProperties.key, keyProperties.salt)
    }
}
